### ISSUE TITLE
Prepare 1.4.0-phetiorc.2 (from master)

### ISSUE THREAD

**Issue by samreid** [phetsims/energy-skate-park-basics#411]
From https://github.com/phetsims/QA/issues/50

We would like to get master stable then publish RC.2, to test the issues linked in qa/50.  Should be done after #381 at least, and @samreid should triage open issues and check status with @jessegreenberg before creating rc.2.

**Comment by samreid** [comment]
If I understand correctly, the plan is to republish ESPB for PhET Brand only, and to defer phet-io until we can do a stable PhET-iO design.  @ariel-phet is that correct?  If so, let's close this issue.

**Comment by jessegreenberg** [comment]
Sim is ready for an RC out of master, Ill wait to hear about which brands are required to build and make the RC task.

**Comment by kathy-phet** [comment]
Since this version of PhET-iO is already available to clients, we should maintain it in sync with what is on the website. That is to say we should republish PhET-iO too.

**Comment by samreid** [comment]
@kathy-phet asked me to see what's changed between master and latest published version.  @zepumph the latest version I see in partners.md is 1.3.1-phet-io.  Has 1.3.2 not been shared with anyone?

Latest dependencies are https://phet-io.colorado.edu/sims/energy-skate-park-basics/1.3.2-phetio/dependencies.json

I tried the API diff tool like so: http://localhost/phet-io-wrappers/api-diff/api-diff.html?url1=https://phet-io.colorado.edu/sims/energy-skate-park-basics/1.3.1-phetio/energy-skate-park-basics_en-phetio.html?production&sim=energy-skate-park-basics&url2=http://localhost/energy-skate-park-basics/energy-skate-park-basics_en.html?brand=phet-io and it failed with 

> Error objects passed over the frame must have a message and a stack

We do not have an easy way to track all issues that were fixed between 2 versions of a sim.  @kathy-phet pointed out that @jonathanolson did a maintenance release for a buggy dragging issue on iOS.  

There have been:
398 commits in axon
113 commits in dot
607 commits in joist
838 commits in scenery
640 commits is sun
195 commits in ESPB
19 commits in ESPB: 1.3-phetio branch (we will need to make sure these made it to master).

To me, it seems sufficient that this is using the new version of studio and fixes the iOS dragging bug--I'm convinced we should move forward with a PhET-iO version.

**Comment by zepumph** [comment]
>@zepumph the latest version I see in partners.md is 1.3.1-phet-io. Has 1.3.2 not been shared with anyone?

I have not shared it with anyone that I know of.

**Comment by samreid** [comment]
Shared and noted in https://github.com/phetsims/phet-io/commit/297a59180857961897ce003f7eb909fb3ba399a9

**Comment by jessegreenberg** [comment]
> I'm convinced we should move forward with a PhET-iO version.

Does that mean that a decision has been made and we should proceed with deploying PhET-iO brand out of master, or are we still considering? @samreid @zepumph what are the next steps for this issue?

**Comment by samreid** [comment]
In discussion with @kathy-phet and @zepumph after today's status meeting, we agreed we will proceed with publishing the PhET-iO version together with the PhET brand version.  I'll leave this self-assigned to look through open issues and do basic PhET-iO testing to make sure this is ready for the QA team.

**Comment by zepumph** [comment]
@samreid I just looked through the phet-io wrapper suite with ESPB and found that the state wrapper is broken (won't change off of the homescreen)

I would also like to mention that I think that the following phet-io issues should be discussed (if not solved) in this release's context before moving forward with qa:
* https://github.com/phetsims/phet-io-wrappers/issues/82 wrappers on the dev site will not work.
*  https://github.com/phetsims/phet-io/issues/1326, a memory leak/ tandem registration problem currently exists in master, and could potentially be band-aided for this release.
* (lesser priority) https://github.com/phetsims/sun/issues/365 for uniformity in the api
* (lesser priority?) https://github.com/phetsims/phet-io/issues/1215 I'm unsure about how this effects ESPB specifically.

**Comment by samreid** [comment]
studio => launch is also broken at the moment, will need to be fixed before QA.  Somewhat related to https://github.com/phetsims/phet-io/issues/1326

**Comment by samreid** [comment]
Studio => launch and the state wrapper are repaired in the preceding commits.

**Comment by samreid** [comment]
I skimmed through the wrappers and they are behaving much better now.  I'm sure QA will find many important things to fix, but it's at least at a point now where it is testable.  @zepumph anything else from you?  @jessegreenberg what else has to happen before we create the RC?

**Comment by jessegreenberg** [comment]
Thanks @samreid! I don't believe anything in this sim specifically.
> I'm sure QA will find many important things to fix, but it's at least at a point now where it is testable

Do you believe a dev test is warranted before proceeding with RC then?

**Comment by samreid** [comment]
> Do you believe a dev test is warranted before proceeding with RC then?

A dev version sounds appropriate.  Let's check with the rest of the team and @zepumph in particular about whether there is any blocking work that would prevent a dev version.  @jessegreenberg will you take the lead on this and create a dev version with `brands=phet,phet-io` when ready?

**Comment by zepumph** [comment]
I am worried about phetsims/phet-io-wrappers#82, and not sure how to proceed in regards to this issue. @samreid can we please discuss this further this afternoon.

**Comment by jessegreenberg** [comment]
https://github.com/phetsims/energy-skate-park-basics/issues/411#issuecomment-396462447 - indeed, just let me know when it is ready.

**Comment by samreid** [comment]
@zepumph do you think we can create a dev test for this sim now?

**Comment by samreid** [comment]
@zepumph pointed out the two linked issues above, once they are closed @jessegreenberg can proceed to create a dev version.

**Comment by zepumph** [comment]
#415 has brought up many real concerns for the phet-io side of this issue, we should wait for that to be taken care of before tackling dev testing.

**Comment by samreid** [comment]
@zepumph and I wrapped up the 3 preceding issues.  @jessegreenberg you are approved to create a dev version for QA for `brands=phet,phet-io`.

**Comment by jessegreenberg** [comment]
Thanks for your work on this @samreid and @zepumph. PhET brand version is being tested in https://github.com/phetsims/QA/issues/134. @zepumph recommended to create an issue for testing the PhET-iO version once https://github.com/phetsims/QA/issues/19 is done. But [both brands were built](https://bayes.colorado.edu/dev/html/energy-skate-park-basics/1.4.0-dev.1/).

**Comment by samreid** [comment]
At Thursday's meeting @kathy-phet recommended to put this re-release on hold.  The PhET-iO ESP/ESPB will require more work and deliberation before we continue.

**Comment by arouinfar** [comment]
This issue is very stale, so I'm going to close it. When we decide to republish ESP(B) with PhET-iO, we can open new issues.
